docker-compose-separate-containers.yml

```diff
@@ -0,0 +1,34 @@
+version: '2'
+services:
+  nginx:
+    image: nginx
+    container_name: nginx
+    ports:
+      - "80:80"
+    volumes:
+      - /tmp/nginx:/etc/nginx/conf.d
+    networks:
+      - proxy-tier
+
+  dockergen:
+    image: jwilder/docker-gen
+    command: -notify-sighup nginx -watch /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
+    volumes_from:
+      - nginx
+    volumes:
+      - /var/run/docker.sock:/tmp/docker.sock:ro
+      - ./:/etc/docker-gen/templates
```

Comment:
A volume is needed to get the nginx.tmpl, but it might as well be a single-file volume. (I didn't realize that that was allowed, but it works.)